app/parse_txs.go

```diff
@@ -24,6 +25,26 @@ func separateTxs(_ client.TxConfig, rawTxs [][]byte) ([][]byte, []core.BlobTx) {
 	return normalTxs, blobTxs
 }
 
+// filterForValidPFBSignature verifies the signatures of the provided PFB transactions. If it is invalid, it
+// drops the transaction.
+func filterForValidPFBSignature(ctx sdk.Context, accountKeeper *keeper.AccountKeeper, txConfig client.TxConfig, txs [][]byte) [][]byte {
+	normalTxs, blobTxs := separateTxs(txConfig, txs)
+
+	// increment the sequences of the standard cosmos-sdk transactions. Panics
+	// from the anteHandler are caught and logged.
+	seqHandler := incrementSequenceAnteHandler(accountKeeper)
+
+	normalTxs, ctx = filterStdTxs(ctx.Logger(), txConfig.TxDecoder(), ctx, seqHandler, normalTxs)
+
+	// check the signatures and increment the sequences of the blob txs,
+	// and filter out any that fail. Panics from the anteHandler are caught and
+	// logged.
+	svHandler := sigVerifyAnteHandler(accountKeeper, txConfig)
+	blobTxs, _ = filterBlobTxs(ctx.Logger(), txConfig.TxDecoder(), ctx, svHandler, blobTxs)
+
+	return append(normalTxs, encodeBlobTxs(blobTxs)...)
```

Comment:
Yeah this is true and I was aware of the redundant encoding/decoding that is being done, however, as you said, I preferred readability over the extra computation cycles this may cause (which I think is very minimal).

I think we could look to change it in a future PR once we have a good gauge on where our bottlenecks are